feat: custom images guide (#595)
stackabletech/documentation: modules/guides/pages/custom-images.adoc

To avoid confusion, do not use the Stackable registry for your custom image, as you cannot push the image there.
You can also keep the versions in the tag, and simply add a suffix to mark your changes.

For example, for a custom image with a MySQL driver added, you might tag your image with `my.corp/stackable/hive:3.1.3-stackable24.3.0-mysql`.

=== Provide the image to the Kubernetes cluster

To deploy containers using this image, the Kubernetes cluster needs to be able to access the image.

You can either upload the image into a custom registry and pull it from there - refer to your registry on how to do this - or make the image available to the Kubernetes cluster directly.
For example, in {kind}[`kind`] you can use the {kind-load-image}[`kind load docker-image`] command to load a local image into the Kind cluster.

=== Use your customized image in your Stacklet definition

To use the customized image in your Stacklet, adjust the xref:concepts:product-image-selection.adoc[] section in your Stacklet configuration.
Use the `custom` field to specify your image:

[source,yaml]
spec:
  image:
    custom: my.corp/stackable/hive:3.1.3-stackable24.3.0-mysql

>     productVersion: 3.3.1

Reviewer: ```suggestion
    productVersion: 3.1.3
```

> Note that you still need to specify the product version was used in the image, as the operator needs to know which version the product is in to configure it correctly.

Reviewer: ```suggestion
Note that you still need to specify the product version was used in the image for the operator  to configure it correctly.
```
